data-service/src/test/java/com/amazonaws/blox/dataservice/integration/DataServiceIntegrationTestBase.java

```diff
@@ -0,0 +1,93 @@
+
+import com.amazonaws.blox.dataservice.api.CreateEnvironmentApi;
+import com.amazonaws.blox.dataservice.api.DataServiceApi;
+import com.amazonaws.blox.dataservice.api.DeleteEnvironmentApi;
+import com.amazonaws.blox.dataservice.api.DescribeEnvironmentApi;
+import com.amazonaws.blox.dataservice.api.DescribeEnvironmentRevisionApi;
+import com.amazonaws.blox.dataservice.api.ListClustersApi;
+import com.amazonaws.blox.dataservice.api.ListEnvironmentsApi;
+import com.amazonaws.blox.dataservice.api.StartDeploymentApi;
+import com.amazonaws.blox.dataservice.mapper.ApiModelMapper;
+import com.amazonaws.blox.dataservice.mapper.EnvironmentMapper;
+import com.amazonaws.blox.dataservice.repository.EnvironmentRepository;
+import com.amazonaws.blox.dataservice.repository.EnvironmentRepositoryDDB;
+import com.amazonaws.blox.dataservice.repository.model.EnvironmentDDBRecord;
+import com.amazonaws.blox.dataservice.repository.model.EnvironmentRevisionDDBRecord;
+import com.amazonaws.blox.dataservicemodel.v1.client.DataService;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.local.embedded.DynamoDBEmbedded;
+import com.amazonaws.services.dynamodbv2.local.shared.access.AmazonDynamoDBLocal;
+import com.amazonaws.services.dynamodbv2.model.CreateTableRequest;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@ContextConfiguration(classes = DataServiceIntegrationTestConfig.class)
+@RunWith(SpringJUnit4ClassRunner.class)
+public abstract class DataServiceIntegrationTestBase {
+  @Autowired private ApiModelMapper apiModelMapper;
+  @Autowired private EnvironmentMapper environmentMapper;
+  private AmazonDynamoDBLocal amazonDynamoDBLocal;
+  protected DataService dataService;
+
+  @Before
+  public void setup() {
+    amazonDynamoDBLocal = DynamoDBEmbedded.create();
+    AmazonDynamoDB amazonDynamoDB = amazonDynamoDBLocal.amazonDynamoDB();
+    DynamoDBMapper dynamoDBMapper = new DynamoDBMapper(amazonDynamoDB);
+    EnvironmentRepository repo = new EnvironmentRepositoryDDB(dynamoDBMapper, environmentMapper);
+    dataService =
+        new DataServiceApi(
+            new CreateEnvironmentApi(apiModelMapper, repo),
+            new DescribeEnvironmentApi(apiModelMapper, repo),
+            new StartDeploymentApi(apiModelMapper, repo),
+            new ListEnvironmentsApi(apiModelMapper, repo),
+            new DescribeEnvironmentRevisionApi(apiModelMapper, repo),
+            new ListClustersApi(apiModelMapper, repo),
+            new DeleteEnvironmentApi(apiModelMapper, repo));
+
+    ProvisionedThroughput throughput =
```

Comment:
The tables setup is duplicated with `EnvironmentRepositoryTestBase`.  We could create a new class, such as `DynamoDBLocalSetup` for the table creation.

Reply:
I was thinking about that, but not sure if it's clear enough to use this common class for both unit test (the repository test) and integ test ( the data service test)